File: ProblemSolving/BinaryTree/LevelOrderTraversal.cpp
/**
 * Definition for a binary tree node.
 * */

#include <iostream>
#include <vector>
#include <queue>

using namespace std;

struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution
{
public:
    vector<vector<int>> levelOrder(TreeNode *root)
    {
        vector<vector<int>> rs;
        queue<TreeNode *> q;
        if (root == NULL)
        {
            return rs;
        }
        q.push(root);
        while (!q.empty())
        {
            int size = q.size();
            vector<int> levelRs;
            for (int i = 0; i < size; i++)
            {
                levelRs.push_back(q.front()->val);
                if (q.front()->left)
                {
                    q.push(q.front()->left);
                }
                if (q.front()->right)
                {
                    q.push(q.front()->right);
                }
                q.pop();
            }
            rs.push_back(levelRs);
        }
        return rs;
    }
};

File: LeetcodePractice/BinaryTree/LevelOrderTraversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        q=deque()
        res=[]
        if(root==None):
            return res
        q.append(root)
        while(len(q)>0):
            size=len(q)
            levelRes=[]
            for i in range(size):
                data=q.popleft()
                levelRes.append(data.val)
                if data.left!=None:
                    q.append(data.left)
                if data.right!=None:
                    q.append(data.right)
            res.append(levelRes)
                
        return res
        